### ISSUE TITLE
Is a web-based Cinderella editor coming?

### ISSUE THREAD

**Issue by q2apro** [CindyJS/CindyJS#701]
As far as I could see, at https://cinderella.de/ there is only the software to be downloaded and then the users have to install it.

It would be fantastic if the same/similar functionality would be provided within the browser. 

Having CindyJS should make it possible.

Out of curiosity: What are the plans to make such an online editor happen?

Thanks!

**Comment by montaga** [comment]
Indeed, we are working on such an editor. However there is no fixed schedule (so it might take some long time until it is finished) and currently only a tiny subset of the functionality of CindyJS is supported in the editor. You can check out the current editor in the master (contributions are, as always, welcome!), and there is a current snapshot on [https://montaga.github.io/cindyeditor/](https://montaga.github.io/cindyeditor/)

**Comment by kortenkamp** [comment]
There are experimental designs. However, I am quite sure that we will not cover the full Cinderella spectrum.

**Comment by q2apro** [comment]
This is a nice beginning!

Our goal is to provide a more advanced geometry editor to our expert users at Mathelounge.de

Here are examples of what they need to build online:

1. https://www.mathelounge.de/529577/schraffierte-viereck-berechnen-aus-der-seite-c?show=529662#a529662

![image](https://user-images.githubusercontent.com/6764548/46345291-adc54180-c64c-11e8-9da3-f778e83f7bb1.png)

2. https://www.mathelounge.de/553743/trapez-seite-lange-c-bestimmen?show=553888#a553888

![image](https://user-images.githubusercontent.com/6764548/46345296-b3bb2280-c64c-11e8-99ef-9b67679ec8b6.png)

It would also be great to not have the need to download a HTML file of CindyJS, but instead the javascript code behind, necessary for the drawing!

Or even a super simple syntax like the one we implemented at: https://www.matheretter.de/rechner/geozeichner or https://www.matheretter.de/geoservant/en/

**Comment by wangyenshu** [comment]
Hello. Complete cinderella2 can run in browser by [cheerpj](https://cheerpj.com/): https://wangyenshu.github.io/cinderella-in-browser/. The web files is placed here: https://github.com/wangyenshu/cinderella-in-browser. Cheerpj supports java up to version 17, so cinderella3 might not be able to run in browser that way. Cheerpj is not open source.

It loads pretty fast if you are using local server. Over Internet it takes about 8 mins because there are too many small files.

<img width="1460" height="918" alt="Image" src="https://github.com/user-attachments/assets/04fdc5f7-81b7-4fed-8962-c7eea20ad6b0" />
